smarts/core/smarts.py

```diff
@@ -712,11 +712,11 @@ def _pybullet_provider_step(self, agent_actions) -> ProviderState:
         return provider_state
 
     def _nondynamic_provider_step(
-        self, agent_actions, step_pybullet: bool
+        self, agent_actions, stepped_pybullet: bool
     ) -> ProviderState:
         self._perform_agent_actions(agent_actions)
```

Comment:
It seems odd to have the `_perform_agent_actions()` occur potentially before or after `_step_pybullet()`

Reply:
It seems to me it always happens before `_step_pybullet()`.  Where do you see it happening before?

Reply:
Nevermind, I understand what you mean now.

Reply:
I'll try a mild refactor to avoid this...

Reply:
Good call.  That would have been an issue in the future.  Just pushed a refactor to attempt to address it.  Please review again!